### ISSUE TITLE
Custom webhook - Unable to send custom query params[BUG]

### ISSUE THREAD

**Issue by prazevj** [opensearch-project/alerting#560]
**What is the bug?**
I have created a new channel under Notifications ( Opensearch version 2.1.0 ) using channel type : "Custom Webhook". Destination is GoAlert ( An open source application for on-call scheduling, automated escalations and notifications ). 
In the message body am trying to send "summary" and "details" of the alert. But, am getting empty values in the GoAlert.

**How can one reproduce the bug?**
1. Go to 'Notifications -> Channel -> Create Channel'
2. Click on 'Custom Webhook as Channel Type"
3. Scroll down to 'Add a GoAlert - Generic API webhook URL'  ( Ref: https://goalert.me/docs/alerting/alerting-from-generic-api/ )
4. Webhook URL - http://<ip-address>:8081/api/v2/generic/incoming?token=<token>
5. Configure a monitor and in the message try adding summary and details as given below. 
{
  "summary": "Alert Summary",
  "details": "Alert details"
}

**What is the expected behavior?**

**What is your host/environment?**
 - OS: [Linux]
 - Version [2.1.0]
 - Plugins

**Do you have any screenshots?**
![image](https://user-images.githubusercontent.com/5722696/190644008-07244555-2048-4b13-86d4-91c641387d12.png)
![image](https://user-images.githubusercontent.com/5722696/190644209-6c054c37-2342-439f-891a-9b22f0327783.png)

**Do you have any additional context?**
In the Opensearch documentation, need more details on how to configure a custom-webhook notification channel. 

https://opensearch.org/docs/latest/notifications-plugin/api/#create-channel-configuration

Only, for slack there is a detailed sample API request and response is available.

**Comment by sbcd90** [comment]
Hi @prazevj, I tested your scenario with a mock http server. https://www.npmjs.com/package/http-server
I created a channel similar to this

& then used the channel to trigger a test message as shown below.
![image](https://user-images.githubusercontent.com/617607/191128229-e11a30d7-6268-4ff2-900e-492a1c3ac651.png)

It worked fine & sent the message body. 
![image](https://user-images.githubusercontent.com/617607/191128272-649f662e-9eb6-4199-b5bd-71639d3ad601.png)

Here is the piece of code which passes `message` body to the request. https://github.com/opensearch-project/notifications/blob/main/notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt#L163

Can you kindly let me know if this is the exact behavior you want? If yes, can you kindly check in `goalert` the exact message you're receiving?

**Comment by prazevj** [comment]
Hi @sbcd90, Thanks for your response. 

In `goalert` am not getting any message for the summary and details (empty summary & details). 

Here in goalert, the summary refers to --> Subject 
and details refers to --> message details.

**Comment by uroskarama** [comment]
@prazevj @praveensameneni 
I have investigated this issue by running GoAlert in a container on my local machine.
Apparently we are sending the body of this POST request to GoAlert's webhook URL in JSON format, which requires using Content-Type: application/json as a header parameter.

Using postman to send requests I discovered that GoAlert doesn't accept this Content-Type: application/json for webhooks. If "summary" and "details" values are passed as URI parameters, or if content type application/x-www-form-urlencoded is passed in the header and values are passed in the body in this format, alerts are created properly.
